pkg/pillar/cmd/diag/diag.go

```diff
@@ -176,9 +176,13 @@ func Run(ps *pubsub.PubSub, loggerArg *logrus.Logger, logArg *base.LogObject, ar
 	}
 	log.Functionf("processed GlobalConfig")
 
-	server, err := os.ReadFile(types.ServerFileName)
-	if err != nil {
-		log.Fatal(err)
+	var server []byte
+	for len(server) == 0 {
```

Comment:
Do we have an infinite loop here if file exists but empty?

Reply:
We sleep and wait for the file to have content, since the following code requires content.

Reply:
You sleep on the error path only, so on empty file you get an infinie loop which burns CPU.

Reply:
The doc says: "ReadFile reads the named file and returns the contents. A successful call returns err == nil, not err == EOF. Because ReadFile reads the whole file, it does not treat an EOF from Read as an error to be reported.", so if my understanding is correct you get an empty bytes read and **no** error, which leads to the busy CPU loop.

Reply:
Ah - got it. Will fix.

Reply:
Fixed